### PR TITLE
Replaced singleLineComments with getCommentSignByLanguage

### PR DESCRIPTION
**Pull request by nausharipov**
#34

### CODE REVIEW

File: lib/src/code/code.dart [hunk 1]
```diff
@@ -26,15 +38,15 @@ class Code {
 
   static List<CodeLine> _textToCodeLines(
     String text,
-    List<String> singleLineComments,
+    String? singleLineComment,
```
Comment (alexeyinkin) [inline lib/src/code/code.dart:41]: Some languages support multiple sequences for single line comments:
https://www.php.net/manual/en/language.basic-syntax.comments.php

So this should still be a list.

File: lib/src/code/code.dart [hunk 2]
```diff
@@ -63,23 +75,26 @@ class Code {
     return result;
   }
 
-  static List<String> _getCommentWords(
-    String str,
-    List<String> singleLineComments,
-  ) {
-    for (final singleLineComment in singleLineComments) {
-      final commentIndex = str.indexOf(singleLineComment);
+  static String? _getCommentSignByLanguage(Mode? language) {
```
Comment (alexeyinkin) [inline lib/src/code/code.dart:78]: This is a sequence and not a sign.

File: lib/src/code/code.dart [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+import 'package:highlight/languages/go.dart';
+import 'package:highlight/languages/java.dart';
+import 'package:highlight/languages/python.dart';
+import 'package:highlight/languages/scala.dart';
 
 import 'code_line.dart';
 import 'text_range.dart';
 import 'tokens.dart';
 
+final List<Mode> slashCommentSignLanguages = [java, dart, go, scala];
```
Comment (alexeyinkin) [inline lib/src/code/code.dart:14]: The cleanest way I can think of is
```dart
const _slashes = '//';
const _hash = '#';

const _singleLineComments = <String, List<String>>{
  {java: [_slashes]},
  {python: [_hash]},
  // ...
};
```

Then just pick from the map without `if`s.

File: lib/src/code_field/code_controller.dart [hunk 4]
```diff
@@ -97,7 +94,7 @@ class CodeController extends TextEditingController {
 
   CodeController({
     String? text,
-    Mode? language,
+    required Mode? language,
```
Comment (alexeyinkin) [inline lib/src/code_field/code_controller.dart:97]: `language` should not be required. It breaks backwards compatibility. If we want others to migrate to our editor, we must support as much use cases as possible.

File: lib/src/code_field/code_controller.dart [hunk 5]
```diff
@@ -317,10 +313,7 @@ class CodeController extends TextEditingController {
         return;
       }
 
-      _lastCode = Code(
-        text: newValue.text,
-        singleLineComments: _singleLineComments,
```
Comment (alexeyinkin) [inline lib/src/code_field/code_controller.dart:322]: Only this line should change. Preserving lines is a virtue.

File: lib/src/code/code.dart [hunk 6]
```diff
@@ -1,20 +1,32 @@
...
+
 class Code {
   final String text;
   final List<CodeLine> lines;
 
   factory Code({
     required String text,
-    required List<String> singleLineComments,
+    required Mode? language,
```
Comment (alexeyinkin) [inline lib/src/code/code.dart:22]: Should not be required.

File: lib/src/code/code.dart [hunk 7]
```diff
@@ -1,20 +1,32 @@
...
   }) {
-    final lines = _textToCodeLines(text, singleLineComments);
-    return Code._(text: text, lines: lines);
```
Comment (alexeyinkin) [inline lib/src/code/code.dart:17]: This line did not have to change.

File: lib/src/code/code.dart [hunk 8]
```diff
@@ -63,23 +75,26 @@ class Code {
...
-      if (commentIndex == -1) {
-        continue;
-      }
+  static List<String> _getCommentWords(String str, String? singleLineComment) {
+    if (singleLineComment != null && str.contains(singleLineComment)) {
+      final commentIndex = str.indexOf(singleLineComment);
 
       return str
           .substring(commentIndex + singleLineComment.length)
           .split(RegExp(r'\s+')); // Split by any whitespaces.
```
Comment (alexeyinkin) [inline lib/src/code/code.dart:94]: You deleted this comment and then re-created it. So in git history it is now yours and not mine. History is very important when looking for what introduced errors. So when you revert the changes, you should squash the commits and publish a new branch.

File: test/src/code/code_test.dart [hunk 9]
```diff
@@ -163,8 +165,8 @@ void main() {
             readonly
             readonly //readonly
             readonly // a readonly b'''
-            '\n\n'
-            '''
+              '\n\n'
+              '''
```
Comment (alexeyinkin) [inline test/src/code/code_test.dart:169]: ?

File: test/src/code/code_test.dart [hunk 10]
```diff
@@ -4,9 +4,11 @@ import 'package:code_text_field/src/code/code.dart';
 import 'package:code_text_field/src/code/code_line.dart';
 import 'package:collection/collection.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:highlight/highlight.dart';
+import 'package:highlight/languages/angelscript.dart';
+import 'package:highlight/languages/java.dart';
 
-const singleLineComments = ['//', '#'];
-
+final Mode language = java;
```
Comment (alexeyinkin) [inline test/src/code/code_test.dart:11]: We prefer inferred typing as it is more readable: `final _language = java`.
Please also make it private.
Please also make `text` private.

File: test/src/code/code_test.dart (hunk 9 above)
Comment (nausharipov) [reply test/src/code/code_test.dart:169]: My formatter works differently. I'll look into it.

File: test/src/code_field/code_controller_test.dart [hunk 11]
```diff
@@ -16,12 +18,17 @@ abc
 ro//readonly
 ''';
 
+final Mode language = java;
```
Comment (alexeyinkin) [inline test/src/code_field/code_controller_test.dart:21]: Private and without type, please.

**Review by alexeyinkin — CHANGES_REQUESTED** [review]
Thank you. Here are the comments.
